### ISSUE TITLE
Model predicts only one class In Binary Classification

### ISSUE THREAD

**Issue by tkrsh** [tensorflow/tensorflow#41952]
Model : 

base_model=tf.keras.applications.EfficientNetB7(input_shape=input_shape,include_top=False, weights='imagenet')

global_average_layer = tf.keras.layers.GlobalAveragePooling2D()
prediction_layer = tf.keras.layers.Dense(1,activation='sigmoid')

model = tf.keras.Sequential([base_model,
  global_average_layer,
  prediction_layer
])

model.compile(optimizer=tf.optimizers.Adam(lr=0.00001),loss='BinaryCrossentropy',metrics=[tf.keras.metrics.AUC()])

history=model.fit(train_data_gen,epochs=1,validation_data=validation_data_gen,callbacks=[lr_scheduler])

sub['Test_Classes']=model.predict_classes(test_data_gen)

Expected Output : 0 1 0 0 0 1 0 1 

Actual Output      : 0 0 0 0 0 0 0 0

**Comment by tkrsh** [comment]
Base Model : base_model=tf.keras.applications.EfficientNetB7(input_shape=input_shape,include_top=False, weights='imagenet')

**Comment by Dynmi** [comment]
@tkrsh  You code seems all right, check the train-data & test-data again?

**Comment by tkrsh** [comment]
train_image_generator=ImageDataGenerator(rescale=1./255)
train_data_gen=train_image_generator.flow_from_dataframe(train,directory=dirctory_train,x_col='image_name',y_col='benign_malignant',class_mode='binary',batch_size=32,target_size=IMG_SHAPE)
LGTM ...

**Comment by Dynmi** [comment]
> train_image_generator=ImageDataGenerator(rescale=1./255)
> train_data_gen=train_image_generator.flow_from_dataframe(train,directory=dirctory_train,x_col='image_name',y_col='benign_malignant',class_mode='binary',batch_size=32,target_size=IMG_SHAPE)
> LGTM ...

I see. Your data source is panda-dataframe, so maybe the problem is in column "benign_malignant". 
I guess-----is there something wrong with the data type?

**Comment by tkrsh** [comment]
I checked train_data_gen it was labelled correctly with 0, 1

**Comment by ravikyram** [comment]
@tkrsh 

Can you please share your colab link or simple standalone code with supporting files to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!
